src/scala/com/github/johnynek/bazel_deps/Writer.scala

```diff
@@ -71,6 +86,54 @@ object Writer {
     }
   }
 
+  def createBuildTargetFile(buildHeader: String, ts: List[Target], tfp: Path, thirdPartyDirectory: DirectoryName): Result[Int] = {
+    (for {
+          b <- IO.exists(tfp.parent)
+          _ <- if (b) IO.const(false) else IO.mkdirs(tfp.parent)
+          _ <- IO.writeUtf8(tfp, createBuildTargetFileContents(buildHeader, ts, thirdPartyDirectory))
+        } yield ())
+      .map(_ => ts.size)
+  }
+
+  def createBuildTargetFileContents(buildHeader: String, ts: List[Target], thirdPartyDirectory: DirectoryName): String = {
+    val separator = "|||"
+    val encodingVersion = 1
+      val lines = ts
+      .sortBy(_.toString)
+      .map { target =>
+        def kv(key: String, value: String, prefix: String = ""): String =
+          s"""$prefix"$key": "$value""""
+
+        def kvOpt(key: String, valueOpt: Option[String], prefix: String = ""): String = valueOpt match {
+          case Some(value) => kv(key, value, prefix)
+          case None => ""
+        }
+
+        def kListV(key: String, values: List[String], prefix: String = ""): String = {
+          val v = values.map { e => "\"" + e + "\""}.mkString(",")
+          s"""$prefix"$key": [$v]"""
+        }
+
+        val targetName = target.name
+        kListV(s"${targetName.path.asString.replace(thirdPartyDirectory.asString, "").stripSuffix("/")}:${targetName.name}", target.listStringEncoding(separator))
```

Comment:
can we lift the first item into a val: `val key = targetName.path.asString....` then use `s"$key:${targetName.name}` or something...

Reply:
👍